include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -1174,6 +1174,20 @@ struct OrtApi {
    * Get the current device id of the GPU execution provider (cuda/tensorrt/rocm).
    */
   ORT_API2_STATUS(GetCurrentGpuDeviceId, _In_ int* device_id);
+
+  /**
+  * Setting intra op thread affnity - only implemented on windows for the branch
+  * affinity string follows format:
+  * logic_processor_id,logic_processor_id;logic_processor_id,processor_id
+  * semicolon isolates configurations among threads, while comma split processors where ith thread expected to attach to.
+  * e.g. 1,2,3;4,5
+  * specifies affinities for two threads, with the 1st thread attach to the 1st, 2nd, and 3rd processor, and 2nd thread to the 4th and 5th processor.
+  * To ease the configuration for thread across many processors, an interval is also allowed:
+  * e.g. 1-8;8-16;17-24
```

Comment:
We should also document how to do this beyond the 1st group in case of Windows since you're planning to encode the group id in the processor id.

Reply:
Adding it, but I thought we are hiding the detail from customers so this API looks purely system-agnostic.